core/server/api_container/server/startosis_engine/kurtosis_starlark_framework/enclave_plan_instruction/enclave_plan_instruction_repository.go

```diff
@@ -0,0 +1,103 @@
+package enclave_plan_instruction
+
+import (
+	"encoding/json"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/database_accessors/enclave_db"
+	"github.com/kurtosis-tech/kurtosis/core/server/api_container/server/startosis_engine/instructions_plan"
+	"github.com/kurtosis-tech/stacktrace"
+	"github.com/sirupsen/logrus"
+	bolt "go.etcd.io/bbolt"
+)
+
+var (
+	enclavePlanInstructionBucketName = []byte("enclave-plan-instruction-repository")
+)
+
+type EnclavePlanInstructionRepository struct {
+	enclaveDb *enclave_db.EnclaveDB
+}
+
+func GetOrCreateNewEnclavePlanInstructionRepository(enclaveDb *enclave_db.EnclaveDB) (*EnclavePlanInstructionRepository, error) {
+	if err := enclaveDb.Update(func(tx *bolt.Tx) error {
+		bucket, err := tx.CreateBucketIfNotExists(enclavePlanInstructionBucketName)
```

Comment:
nit: you can move this createBucketIfNotExists in the `Save` function, this will simplify the interface of this function by not returning an error